app/views/search/show.html.erb

```diff
@@ -0,0 +1,11 @@
+<%= @gym.name %><br/>
+<%= @gym.address %><br/>
+<%= @gym.phone %>
+
+<%= form_with url: gyms_path, method: :post, local: true do |f|  %>
+<%= f.hidden_field :user_id, value: current_user.id %>
```

Comment:
I think if you are passing these as params in the path helper, you don't need the hidden fields too

Reply:
Removed commented out button that wasn't functional.